.github/workflows/main.yml

```diff
@@ -0,0 +1,191 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "v*.*.*"
+  pull_request:
+    branches:
+      - master
+
+env:
+  AWS_REGION: "us-west-2"
+  DEFAULT_PYTHON_VERSION: "3.11"
+
+permissions:
+  id-token: write
+  contents: write
+  pull-requests: write
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "${{ env.DEFAULT_PYTHON_VERSION }}"
+      - name: build wheels
+        uses: PyO3/maturin-action@v1
+        with:
+          args: --release --out ../../dist --interpreter '3.9 3.10 3.11'
+          rust-toolchain: stable
+          docker-options: -e CI
+          working-directory: src/python_wrapper
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+      - name: Syntax Check
+        working-directory: src/python_wrapper
+        run: cargo fmt --all --check
+      - name: Cargo Tests
+        if: success() || failure()
```

Comment:
are there other options besides success and failure ?

Reply:
I was also puzzled by this - https://stackoverflow.com/a/58859404 explained it: 
> If you don't want the function to run when you manually cancel a job [...]

Reply:
Yes, this is an old trick to do `allow_failure: true` as in Gitlab CI (run tests even if the syntax check failed). I was not using `'!canceled()'` as GH VS Code extension will try to reformat to `"!canceled()"` and then complaints that it should not start with `"`...

Reply:
yeah. anything that requires this sort of escaping looks weird and probably behaves weird